punishments.js

```diff
@@ -1486,11 +1483,12 @@ Punishments.isRoomBanned = function (user, roomid) {
  * options.publicOnly will make this only return public room punishments.
  * options.checkIps will also check the IP of the user for IP-based punishments.
  *
- * @param {User | string} user
+ * @param {User? | string} user
  * @param {Object?} options
  * @return {Array}
  */
 Punishments.getRoomPunishments = function (user, options) {
+	if (typeof user === 'string') user = Users(user);
```

Comment:
Also this is a bug, we intentionally don't do this because `getRoomPunishments` is supposed to be able to find punishments for offline users.